felix/fv/service_loop_prevention_test.go

```diff
@@ -223,14 +256,22 @@ var _ = infrastructure.DatastoreDescribe("service loop prevention; with 2 nodes"
 		})
 
 		// Expect to see empty cali-cidr-block chains.  (Allowing time for a Felix restart.)
-		for _, felix := range tc.Felixes {
-			Eventually(getCIDRBlockRules(felix, "iptables-save"), "8s", "0.5s").Should(BeEmpty())
-			Eventually(getCIDRBlockRules(felix, "ip6tables-save"), "8s", "0.5s").Should(BeEmpty())
+		for id, felix := range tc.Felixes {
+			if BPFMode() {
+				curId := tcProgID[id]
+				Eventually(func() int {
+					tcProgID[id] = getTcProgID(felix, "eth0")
+					return tcProgID[id]
+				}, "15s", "1s").ShouldNot(BeNumerically("==", curId))
```

Comment:
shouldn't you rather test that the blackhore rules are cleared?

Reply:
Blackhole routes are cleared only when the BGPConfig is deleted. There is ```ServiceLoopPrevention``` felix config, which is modified in this test. Config change triggers a felix restart. I tried using ```ensureAllNodesBPFProgramsAttached``` but that did not work well because the test did not wait for the felix to restart. Is there a way for the test to know if felix is restarting?

Reply:
Found a way.